docs/en/testing/introduction.md

```diff
@@ -109,61 +124,78 @@ Assertion templates provide a way to create a base assertion that allow parts of
 
 -   Given a widget that renders output differently based on property values:
 
-> src/widgets/Profile.ts
+> src/widgets/Profile.tsx
 
 ```tsx
+import { create, tsx } from '@dojo/framework/core/vdom';
+
+import * as css from './Profile.m.css';
+
 export interface ProfileProperties {
 	username?: string;
 }
 
-export default class Profile extends WidgetBase<ProfileProperties> {
-	protected render() {
-		const { username } = this.properties;
-		return v('h1', { classes: [css.root] }, [`Welcome ${username || 'Stranger'}!`]);
-	}
-}
+const factory = create().properties<ProfileProperties>();
+
+const Profile = factory(function Profile({ properties }) {
+	const { username } = properties();
+	return <h1 classes={[css.root]}>{`Welcome ${username || 'Stranger'}!`}</h1>;
+});
+
+export default Profile;
 ```
 
 -   Create an assertion template using `@dojo/framework/testing/assertionTemplate`
 
-> tests/unit/widgets/Profile.ts
+> tests/unit/widgets/Profile.tsx
 
 ```ts
+const { describe, it } = intern.getInterface('bdd');
+import { tsx } from '@dojo/framework/core/vdom';
+import assertionTemplate from '@dojo/framework/testing/assertionTemplate';
+import harness from '@dojo/framework/testing/harness';
+
+import Profile from '../../../src/widgets/Profile';
+import * as css from '../../../src/widgets/Profile.m.css';
+
 // Create an assertion
-const profileAssertion = assertionTemplate(() =>
-	v('h1', { classes: [css.root], '~key': 'welcome' }, ['Welcome Stranger!'])
-);
+const profileAssertion = assertionTemplate(() => (
+	<h1 classes={[css.root]} assertion-key="welcome">
+		Welcome Stranger!
+	</h1>
+));
 
 describe('Profile', () => {
 	it('default renders correctly', () => {
-		const h = harness(() => w(Profile, {}));
+		const h = harness(() => <Profile />);
 		// Test against my base assertion
 		h.expect(profileAssertion);
 	});
 });
 ```
 
-A value can be provided to any virtual DOM node under test using the `~key` properties defined in the assertion template. In `.tsx` this would be the `assertion-key` attribute.
+A value can be provided to any virtual DOM node under test using `assertion-key` properties defined in the assertion template. Note: when using class-based widgets, the `~key` property serves the same purpose.
```

Comment:
This isn't only for class-based widgets - its usable when using the functional `w()` and `v()` APIs

Reply:
changed wording to improve accuracy